ios/brave-ios/Sources/BraveWallet/Crypto/SelectAccountTokenView.swift

```diff
@@ -138,11 +138,22 @@ struct SelectAccountTokenView: View {
         },
         header: {
           WalletListHeaderView {
-            CopyAddressHeader(
-              displayText: accountSection.account.accountNameDisplay,
-              account: accountSection.account,
-              btcAccountInfo: accountSection.bitcoinAccountInfo
-            )
+            if let btcAccountInfo = accountSection.bitcoinAccountInfo {
+              CopyAddressHeader(
+                displayText: accountSection.account.accountNameDisplay,
+                accountInfo: .btc(btcAccountInfo)
+              )
+            } else if let zecAccountInfo = accountSection.zcashAccountInfo {
+              CopyAddressHeader(
+                displayText: accountSection.account.accountNameDisplay,
+                accountInfo: .zec(zecAccountInfo)
+              )
+            } else {
+              CopyAddressHeader(
+                displayText: accountSection.account.accountNameDisplay,
+                accountInfo: .regular(accountSection.account)
+              )
+            }
```

Comment:
Would prefer having a computed property to just return the `CopyAddressHeader.AccountInfo` as that's all that changes. Maybe also makes sense to move the `CopyAddressHeader.AccountInfo` outside of `CopyAddressHeader`? Could be useful elsewhere :).

```    
struct AccountSection: Equatable, Identifiable {
    <...>
    var accountInfo: AccountInfo {
      if let bitcoinAccountInfo {
        return .btc(bitcoinAccountInfo)
      } else if let zcashAccountInfo {
        return .zec(zcashAccountInfo)
      } else {
        return .regular(account)
      }
    }
} // AccountSection
```